### PR TITLE
#17 Documentation

### LINKED ISSUE #17: Update documentation to give clear examples of how it all works
Lets not make this a true open source utility :D

### PR DESCRIPTION
**Pull request by brettminnie**
Will close #17

### CODE REVIEW

**Comment by brettminnie** [comment]
@fpietka & @eddiejaoude please sanity check my English here and let me know if I have missed anything

File: README.md [hunk 1]
```diff
@@ -3,22 +3,50 @@
 [![Build Status](https://travis-ci.org/brettminnie/breakfast-serializer.svg)](https://travis-ci.org/brettminnie/breakfast-serializer)
 [![Scrutinizer Code Quality](https://scrutinizer-ci.com/g/brettminnie/breakfast-serializer/badges/quality-score.png?b=develop)](https://scrutinizer-ci.com/g/brettminnie/breakfast-serializer/?branch=develop)
 [![Dependency Status](https://www.versioneye.com/user/projects/55378b007f43bcd88900033d/badge.svg?style=flat)](https://www.versioneye.com/user/projects/55378b007f43bcd88900033d)
+[![Coverage Status](https://coveralls.io/repos/brettminnie/breakfast-serializer/badge.svg?branch=master&service=github)](https://coveralls.io/github/brettminnie/breakfast-serializer?branch=master)
 
-A (hopefully) backwards compatible replacement for the other well known serializer
+A replacement for the other well known serializer, initially we where aiming for backwards compatibility, however it seemed
```
Comment (fpietka) [inline README.md]: `we were aiming for`

Comment (brettminnie) [inline README.md]: :+1: Fixed and clarified end of sentence

File: .travis.yml [hunk 2]
```diff
@@ -7,15 +7,20 @@ matrix:
...
-  - 7.0 
+  - 7.0
+
+install:
+  - composer require satooshi/php-coveralls:~0.6@stable
 
 before_script:
     - composer self-update
     - composer install --dev
+    - mkdir -p build/logs
```
Comment (eddiejaoude) [inline .travis.yml]: Not a big deal, but the indentation isn't the same as above.

Comment (brettminnie) [inline .travis.yml]: :+1:

File: README.md [hunk 3]
```diff
@@ -3,22 +3,50 @@
...
+ - [x] Mapping of properties to alternate names and back again
+ - [x] Excluding of properties from serialization and ignoring them on deserialization
+ - [x] Simple YAML config format (Yay no slow php annotations!)
+ 
+Quick and Dirty Example
 ```php
    
-   //To retrieve json of an object
-   $jsonData = BDBStudios\BreakfastSerializer\SerializerFactory::getSerializer()->serialize($myClass);
+   // To retrieve the json representation json of an object
```
Comment (fpietka) [inline README.md]: This is alternating between 3 and 4 spaces

File: README.md [hunk 4]
```diff
@@ -3,22 +3,50 @@
...
+    $jsonData = BDBStudios\BreakfastSerializer\SerializerFactory::getSerializer()
+        ->serialize($myClass);
    
    
-   //To unserialize
-   $myUnserializedObject = BDBStudios\BreakfastSerializer\SerializerFactory:::getSerializer()->deserialize($jsonData);
+   // To unserialize
+   $myClass = BDBStudios\BreakfastSerializer\SerializerFactory:::getSerializer()
```
Comment (fpietka) [inline README.md]: Too many `:` here

**Comment by fpietka** [comment]
I linked just a minor issue, although it is also on the `develop` branch...

**Comment by brettminnie** [comment]
@fpietka @eddiejaoude rebased against develop, can this be merged please

**Comment by eddiejaoude** [comment]
Scrutinizer is failing on Code Coverage - does it need to wait for more time?

screen shot 2015-07-25 at 07 55 30

**Comment by fpietka** [comment]
Travis seems to be fine.
I must say that in my company's project, we made scruti build itself because we had too much issue with it.

**Comment by brettminnie** [comment]
Yeah, ignore Scrutinser. Coverage will be on coveralls now post this merge

**Comment by brettminnie** [comment]
@eddiejaoude waiting on you, Scrutinizer woke up

**Comment by brettminnie** [comment]
@eddiejaoude This has had the bugfixes merged into it from Develop and can be merged in now ignoring scrutinizer being all laggy

File: README.md [hunk 5]
```diff
@@ -3,29 +3,57 @@
...
-A (hopefully) backwards compatible replacement for the other well known serializer
+A replacement for the other well known serializer, initially we were aiming for backwards compatibility, however it seemed
+more appropriate to develop a lightweight alternative. We are forgoing the depth of features now to offer something that is
+easy to configure and works well without any configuration.
 
-Currently will only support json and do a full depth/breadth recursion, we'll build on this slowly
+### Installation
 
-Example
+`$ composer require brettminnie/breakfast-serializer`
```
Comment (eddiejaoude) [inline README.md]: Might be worth putting a version on the end of this? `$ composer require brettminnie/breakfast-serializer ...VERSION...`

File: documentation/remapping.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    </span>
+</nav>
+
+### Remapping Data
+
+Sometimes we may wish to rename a property when being serialized as the returned name may either class with part of the 
+api consumer or the consumer may not understand your naming conventions.
+
+Each class that requires properties to be remapped can be set up in a yaml file in the config directory. See 
+[mappings.yml](../config/mappings/mappings.yml) for a simple example from our test suite. This is commented out as
```
Comment (eddiejaoude) [inline documentation/remapping.md]: You could use **Puli** here for nicer paths http://puli.io

File: README.md (hunk 5 above)
Comment (brettminnie) [inline README.md]: :+1:

File: documentation/remapping.md (hunk 6 above)
Comment (brettminnie) [inline documentation/remapping.md]: will look at it cheers

**Pull request merged**